### PR TITLE
add script to auto-generate version-specific pages

### PR DESCRIPTION
**Pull request by boegel**
requires:
* https://github.com/easybuilders/easybuild-framework/pull/4117

The `update-version-specific-docs.sh` can be used to auto-generate the pages under `version-specific/` .
To be clear: the intention is to run this script manually, as a part of the release process (see also [here](https://github.com/easybuilders/easybuild/wiki/Release-management)).

There are a couple of minor issues that we should try and resolve, but overall this should be good enough as a first iteration to replace what we currently have at https://easybuild.readthedocs.io/en/latest/version-specific ; we can open issues on whatever small problems we find along the way, and fix them in follow-up PRs.

The included updated `.md` pages under `version-specific` correspond to the current `develop` branches of EasyBuild (pre-v4.7.0)

### CODE REVIEW

File: docs/version-specific/easyblocks.md [hunk 1]
```diff
@@ -1,16 +1,402 @@
 # List of easyblocks {: #vsd_list_easyblocks }
 
-## Overview of generic easyblocks {: #generic_easyblocks }
```
Comment (boegel) [inline docs/version-specific/easyblocks.md:3]: I've moved the `#generic_easyblocks` anchor to `generic-easyblocks.md`

**Comment by boegel** [comment]
@branfosj I'm hoping we can ignore the failing MarkDown linting and codespell tests for now, and get this merged as is, since https://github.com/easybuilders/easybuild-framework/pull/4117 is blocking the release of EasyBuild v4.7.0...

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -19,7 +19,8 @@ jobs:
         with:
           check_filenames: true
           ignore_words_list: atleast,ninjs,simpy,proovread,namd,precice
-          exclude_file: docs/js/asciinema-player-2.6.1.js
+          # filter out version-specific/* for now, until we fix all issues that mdl trips over
+          exclude_file: 'docs/js/asciinema-player-2.6.1.js|docs/version-specific/*'
```
Comment (branfosj) [inline .github/workflows/test.yml:23]:
```suggestion
          skip: './docs/js/asciinema-player-2.6.1.js, ./docs/version-specific'
```

Exclude file is the wrong option.

File: .github/workflows/markdown-linting.yml [hunk 3]
```diff
@@ -12,4 +12,5 @@ jobs:
       - name: MarkdownLint mdl Action
         uses: actionshub/markdownlint@2.0.2
         with:
-          filesToIgnoreRegex: "docs/demos/.*"
+          # filter out version-specific/* for now, until we fix all issues that mdl trips over
+          filesToIgnoreRegex: "docs/demos/.*|docs/version-specific/.*"
```
Comment (branfosj) [inline .github/workflows/markdown-linting.yml:16]:
```suggestion
          filesToIgnoreRegex: "(docs/demos/.*|docs/version-specific/.*)"
```

We might need to escape the / as \/

**Comment by branfosj** [comment]
For the spelling:
* just skip the supported software page (as the errors being detected are not ours!)
* requires https://github.com/easybuilders/easybuild-framework/pull/4154 to fix the last few issues

**Review by branfosj — APPROVED** [review]

**Pull request merged**